neptune-core/src/application/loops/connect_to_peers.rs

```diff
@@ -339,17 +341,27 @@ where
     >::new(length_delimited, SymmetricalBincode::default());
 
     // Complete Neptune handshake
-    let Some(PeerMessage::Handshake {
-        magic_value,
-        data: peer_handshake_data,
-    }) = peer.try_next().await?
-    else {
-        bail!("Didn't get handshake on connection attempt, from {peer_address}");
+    let maybe_msg = timeout(Duration::from_secs(4), peer.try_next()).await;
```

Comment:
Make the timeout configurable to allow simple adaptation under bad network conditions?

Reply:
OK. Defaults to four seconds.